common/src/autogluon/common/utils/s3_utils.py

```diff
@@ -29,3 +29,47 @@ def delete_s3_prefix(bucket, prefix):
 
     if len(delete_keys['Objects']) != 0:
         s3.meta.client.delete_objects(Bucket=bucket, Delete=delete_keys)
+        
+
+def download_s3_folder(bucket, prefix, local_path="."):
+    """
+    This util function downloads a s3 folder and maintain its structure.
+    For example, assuming bucket = bar and prefix = foo, and the bucket structure looks like this
```

Comment:
I think I didn't get the question here.
So if you have `s3://bar/foo/a.txt`, you provide `bucket=bar` and `prefix=foo/a.txt`, then it will download `a.txt`
if you have `s3://bar/fooa/foo/a.txt`, you provide `bucket=bar` and `prefix=fooa/foo/a.txt`, then it will download the `a.txt` located in this subdir